pyproject.toml

```diff
@@ -7,12 +7,13 @@ repository = "https://github.com/eliorc/node2vec"
 license = "MIT"
 
 [tool.poetry.dependencies]
-python = "^3.8"
-networkx = "^2.5"
+python = "^3.11"
+networkx = "^3.3"
 gensim = "^4.1.2"
 numpy = "^1.19.5"
 tqdm = "^4.55.1"
 joblib = "^1.1.0"
+scipy = "1.12"
```

Comment:
There has been a lot of chaos around the current gensim 4.3.3 release and its use of scipy. My suggestion would be to not pin the scipy version here (as gensim is the only one it depends on) but instead update the gensim version to `^4.3.3`.